src/Reflection/ClassReflection.php

```diff
@@ -425,7 +427,7 @@ public function hasProperty(string $propertyName): bool
 		}
 
 		foreach ($this->propertiesClassReflectionExtensions as $i => $extension) {
-			if ($i > 0 && !$this->allowsDynamicPropertiesExtensions()) {
+			if ($i > 1 && !$this->allowsDynamicPropertiesExtensions()) {
```

Comment:
This looks like some magic. I'd want to inject the new extensions into ClassReflection and call them regardless of `allowsDynamicPropertiesExtensions`.